test/mustachio/builder_test.dart

```diff
@@ -97,46 +119,34 @@ import 'package:mustachio/annotations.dart';
 class Foo {}
 class Bar {}
 ''',
-    }, outputs: {
-      'foo|lib/foo.renderers.dart': _containsAllOf(
-          // The requested 'renderFoo' function
-          'String renderFoo(Foo context, List<MustachioNode> ast)',
-          // The renderer class for Foo
-          'class _Renderer_Foo extends RendererBase<Foo>',
-          // The requested 'renderBar' function
-          'String renderBar(Bar context, List<MustachioNode> ast)',
-          // The renderer class for Bar
-          'class _Renderer_Bar extends RendererBase<Bar>')
     });
+    var renderersLibrary = await resolveGeneratedLibrary(writer);
+
+    expect(renderersLibrary.getTopLevelFunction('renderFoo'), isNotNull);
+    expect(renderersLibrary.getTopLevelFunction('renderBar'), isNotNull);
+    expect(renderersLibrary.getType('_Renderer_Foo'), isNotNull);
+    expect(renderersLibrary.getType('_Renderer_Bar'), isNotNull);
   });
 
-  test('builds a renderer for a class which extends another class', () async {
-    await testBuilder(mustachioBuilder(BuilderOptions({})), {
-      ..._annotationsAsset,
+  test('builds a renderer for a class which is extended by a rendered class',
+      () async {
+    await testMustachioBuilder({
       'foo|lib/foo.dart': '''
-@Renderer(#renderFoo, Context<Foo>(), 'bar.html.mustache')
-library foo;
-import 'package:mustachio/annotations.dart';
-
+$_libraryFrontMatter
 class FooBase {}
-
 class Foo extends FooBase {}
 ''',
-    }, outputs: {
-      'foo|lib/foo.renderers.dart': _containsAllOf(
-          'String _render_FooBase(FooBase context, List<MustachioNode> ast)',
-          'class _Renderer_FooBase extends RendererBase<FooBase>')
     });
+    var renderersLibrary = await resolveGeneratedLibrary(writer);
+
+    expect(renderersLibrary.getTopLevelFunction('_render_FooBase'), isNotNull);
+    expect(renderersLibrary.getType('_Renderer_FooBase'), isNotNull);
   });
 
   test('builds a renderer for a generic type', () async {
-    await testBuilder(mustachioBuilder(BuilderOptions({})), {
-      ..._annotationsAsset,
+    await testMustachioBuilder({
       'foo|lib/foo.dart': '''
-@Renderer(#renderFoo, Context<Foo>(), 'bar.html.mustache')
-library foo;
-import 'package:mustachio/annotations.dart';
-
+$_libraryFrontMatter
```

Comment:
optional:  might be cleaner to refactor as a parameter to testMustachioBuilder that it adds in itself

Reply:
Great idea! Done